packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts

```diff
@@ -7,3 +9,178 @@ export const generateDefaultRDSExpression = (): string => {
   const exp = methodCall(ref('util.unauthorized'));
   return printBlock('Default RDS Auth Resolver')(compoundExpression([exp, toJson(obj({}))]));
 };
+
+export const generateAuthRulesFromRoles = (roles: Array<RoleDefinition>, fields: Readonly<FieldDefinitionNode[]>): Expression[] => {
+  const expressions = [];
+  expressions.push(
+    qref(methodCall(ref('ctx.stash.put'), str('hasAuth'), bool(true))),
+    set(ref('authRules'), list([])),
+  );
+  const fieldNames = fields.map((field) => field.name.value);
+  roles.forEach((role) => {
+    expressions.push(convertAuthRoleToVtl(role, fieldNames));
+  });
+  return expressions;
+};
+
+const convertAuthRoleToVtl = (role: RoleDefinition, fields: string[]): Expression => {
+  const allowedFields = getAllowedFields(role, fields).map((field) => str(field));
+
+  // Api Key
+  if (role.provider === 'apiKey') {
+    return qref(
```

Comment:
There is too much complexity in this method, can we break this down based on the provider and/or strategy?